### DISCUSSION TITLE
Questions of training and Inference of EcRam

### DISCUSSION THREAD

**Discussion by xsu0960** [IBM/aihwkit#665] [Q&A]
I tried using rpu_config = EcRamPreset() to complete the HWA training, and it seems to have worked well.  (1) But for the inference part, should I still use this rpu_config, or should I specifically define an InferenceRPUConfig()? If I use the InferenceRPUConfig(), how could I put EcRAM device information into this InferenceRPUConfig()? (2) Another question is, is there a method for me to change the g_min and g_max of EcRAM? I don't find these two parameters in EcRamPreset(). Thanks!

**Comment by kaoutar55** [comment]
For the inference hardware-aware training part, you should use the inferenceRPUConfig. @maljoras is there a way to make the g_min and g_max configurable for this preset?

**Reply by xsu0960** [reply]
Thanks for your response! Now, I am trying to do the training and inference simulation with EcRam devices with different memory windows. If you can provide some strategies,  I would be grateful.  

If I use InferenceRPUConfig(), how do I put the EcRamPreset into it? And I also see the noise_model parameter in InferenceRPUConfig(). Because there is no EcRamNoiseModel in aihwkit, which noise model should I use for EcRAM simulation?

Thanks.
